cmd/frontend/internal/cli/autoupgrade.go

```diff
@@ -198,20 +210,27 @@ func runMigration(ctx context.Context,
 	)
 }
 
+type dialer func(_ *observation.Context, dsn string, appName string) (*sql.DB, error)
+
 // performs the role of `migrator up`, applying any migrations in the patch versions between the minor version we're at (that `upgrade` brings you to)
 // and the patch version we desire to be at.
 func finalMileMigrations(obsvCtx *observation.Context) error {
 	dsns, err := postgresdsn.DSNsBySchema(schemas.SchemaNames)
 	if err != nil {
 		return err
 	}
-	schemas := []string{"frontend", "codeintel", "codeinsights"}
-	for i, fn := range []func(_ *observation.Context, dsn string, appName string) (*sql.DB, error){
-		connections.MigrateNewFrontendDB, connections.MigrateNewCodeIntelDB, connections.MigrateNewCodeInsightsDB,
-	} {
-		sqlDB, err := fn(obsvCtx, dsns[schemas[i]], "frontend")
+
+	migratorsBySchema := map[string]dialer{
+		"frontend":     connections.MigrateNewFrontendDB,
+		"codeintel":    connections.MigrateNewCodeIntelDB,
+		"codeinsights": connections.MigrateNewCodeInsightsDB,
+	}
+	for schema, migrateLastMile := range migratorsBySchema {
+		obsvCtx.Logger.Info("Running last-mile migrations", log.String("schema", schema))
+
+		sqlDB, err := migrateLastMile(obsvCtx, dsns[schema], "frontend")
```

Comment:
```suggestion
		sqlDB, err := migrateLastMile(obsvCtx, dsns[schema], appName)
```
no big deal anymore (mostly just matters for the old connections block at the bottom) but just nice to have it named anyway